src/ucs/config/parser.c

```diff
@@ -1238,26 +1238,48 @@ static void ucs_config_parser_mark_env_var_used(const char *name, int *added)
     pthread_mutex_unlock(&ucs_config_parser_env_vars_hash_lock);
 }
 
-static int ucs_config_parse_config_file_line(void *user, const char *section,
-                                             const char *name,
-                                             const char *value)
+static khiter_t ucs_config_get_iter_from_config_file(const char *name)
 {
-    int result;
-    khiter_t iter;
+    return kh_get(ucs_config_map, &ucs_config_file_vars, name);
+}
 
-    iter = kh_put(ucs_config_map, &ucs_config_file_vars,
-                  ucs_strdup(name, "parse_config_file_line name"), &result);
+static char *ucs_config_get_value_from_config_file(const char *name)
+{
+    khiter_t iter = ucs_config_get_iter_from_config_file(name);
 
-    if (result < 0) {
-        ucs_error("Inserting '%s' to config map failed", name);
-        return 0;
-    } else if (result == 0) {
-        ucs_error("Found duplicate '%s' in config map", name);
-        return 0;
+    if (iter == kh_end(&ucs_config_file_vars)) {
+        return NULL;
+    }
+
+    return kh_val(&ucs_config_file_vars, iter);
+}
+
+
+static int
+ucs_config_parse_config_file_line(void *override, const char *section,
+                                  const char *name, const char *value)
+{
+    int result;
+    khiter_t iter = ucs_config_get_iter_from_config_file(name);
+
+    if (iter != kh_end(&ucs_config_file_vars)) {
+        if (*(int*) override) {
+            ucs_free(ucs_config_get_value_from_config_file(name));
+        } else {
+            ucs_error("found duplicate '%s' in config map", name);
+            return 0;
+        }
+    } else {
+        iter = kh_put(ucs_config_map, &ucs_config_file_vars,
+                      ucs_strdup(name, "parse_config_file_line name"), &result);
+        if (result == UCS_KH_PUT_FAILED) {
```

Comment:
why need another kh_put? we already have iter

Reply:
We reach this code only if the key is not in the table -> need to insert it:
```
if (iter != kh_end(&ucs_config_file_vars)) {
...
} else {
// HERE
```